Add new subcommand "scoop diag"

r15ch13 opened the pull request:
#### Description
This introduces the new subcommand `scoop diag`.
It returns information about the Scoop environment using the Ini file format that can be posted on a GitHub issue (including markdown codeblock).
Sensitive content and the username gets automatically removed.

Is there any other information that could be added to this?

Example output of my current system:
```ini
[PowerShell]
Path         = C:\Users\<username>\scoop\apps\pwsh\current\pwsh.exe
Version      = 7.3.4
Edition      = Core
Architecture = 64bit
RunAsAdmin   = False
[Git]
Path         = C:\Users\<username>\scoop\apps\git\current\mingw64\bin\git.exe
Version      = git version 2.41.0.windows.1
[Environment]
HTTPS_PROXY  = <redacted>
HTTP_PROXY   = <redacted>
[Scoop]
Outdated     = True
OnHold       = False
Config       = C:\Users\<username>\.config\scoop\config.json
CoreRoot     = X:\~projects\~powershell\scoop
ScoopDir     = C:\Users\<username>\scoop
GlobalDir    = C:\ProgramData\scoop
CacheDir     = C:\Users\<username>\scoop\cache
[Config]
debug        = True
last_update  = 06/12/2023 20:47:17
gh_token     = <redacted>
virustotal_api_key = <redacted>
aria2-warning-enabled = False
scoop_branch = develop
scoop_repo   = https://github.com/ScoopInstaller/Scoop
```


#### Motivation and Context
Get more information from users when they are opening issues.
#### Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [ ] I have updated the documentation accordingly.
- [ ] ~I have updated the tests accordingly.~
- [ ] I have added an entry in the CHANGELOG.

HUMORCE commented:
Developer mode and other missing entries in `scoop checkup` should be added also, IMO.

HUMORCE commented:
Consider add it as a subcommand of scoop-checkup(`scoop checkup diag`), or a verbose/special output.(scoop checkup --verbose|diag...)

r15ch13 commented:
![image](https://github.com/ScoopInstaller/Scoop/assets/432127/adf0ea7f-8134-4b3e-8a54-6b18d614cf96)
```ini
[PowerShell]
Path         = $env:USERPROFILE\scoop\apps\pwsh\current\pwsh.exe
Version      = 7.3.7
Edition      = Core
Architecture = 64bit
RunAsAdmin   = False
Parent       = Code.exe
[Helpers]
GitPath      = $env:USERPROFILE\scoop\apps\git\current\mingw64\bin\git.exe
GitVersion   = git version 2.42.0.windows.2
Zip          = True
Lessmsi      = True
Innounp      = True
Dark         = True
Aria2        = True
Zstd         = False
[Environment]
SCOOP        = $env:USERPROFILE\scoop
[Scoop]
Outdated     = True
OnHold       = False
Config       = $env:USERPROFILE\.config\scoop\config.json
CoreRoot     = X:\~projects\~powershell\scoop
ScoopDir     = $env:USERPROFILE\scoop
CacheDir     = $env:USERPROFILE\scoop\cache
GlobalDir    = C:\ProgramData\scoop
[Config]
debug                 = True
last_update           = 10/09/2023 15:00:14
analytics_id          = <redacted>
aria2-warning-enabled = False
scoop_repo            = https://github.com/ScoopInstaller/Scoop
scoop_branch          = develop
[Windows]
DeveloperMode     = True
LongPathsEnabled  = False
ScoopDirFormat    = NTFS
GlobalDirFormat   = NTFS
WindowsDefender   = Running
```

Added the entries from `scoop checkup` and some color. Refactored `is_admin` and the check functions.

niheaven commented:
Remove the Markdown `ini` tag? If used in console, that's strange IMO.
